Include open buffers as match candidates.

pavoljuhas opened the pull request:
Vim buffers that are not yet in `:oldfiles` will be there in
the next session.  Let's include them right away.  Avoid
mental exercise figuring if the desired file is old or not.

pavoljuhas committed:
Include open buffers as match candidates.

Vim buffers that are not yet in `:oldfiles` will be there in
the next session.  Let's include them right away.  Avoid
mental exercise figuring if the desired file is old or not.

pavoljuhas commented:
@tyru - would you mind taking a quick look at this change?

tyru commented on plugin/oldfilesearch.vim:
```diff
@@ -75,6 +75,20 @@ function! s:GetOldFiles(patterns)
+            continue
+        endif
+        " skip buffers that are already candidates.
+        let bfull = fnamemodify(bufname(l:b), ':p')
```
it seems you should check whether `bufname(l:b)` is empty.
`fnamemodify('', ':p')` is converted to current directory.
maybe it is not what we want, isn't it?

pavoljuhas replied:
Good catch, thank you!  The directory happens to get excluded later, because it is not a file.  It is still better to fix this up - ce3645e95a2fbb97d60acb47601c901449248172.

tyru commented:
Checking out and tried it and saw the source code change,
added comment that I am concerned about.

pavoljuhas committed:
Exclude unnamed buffers explicitly.

Full path of an unnamed buffer expands to the current directory,
which leaks to the candidate list and is later dropped because
it is not a file.  Refactor to explicitly ignore unnamed buffers.
(Takuya Fujiwara)

tyru commented:
The latest changed script looks good to me 😄

tyru commented:
Wow, I did not notice that your commit, comment, my comment are within 1 or 2 minutes...

pavoljuhas commented:
Thank you.
